src/ast/mod.rs

```diff
@@ -450,6 +450,8 @@ pub type Command = GenericCommand<Symbol, Symbol>;
 
 pub type Subsume = bool;
 
+pub type Subdatatypes = Result<Vec<Variant>, (Symbol, Vec<Expr>)>;
```

Comment:
Are you using `Result` to separate sorts vs datatypes? I think it might be clearer if you have a custom enum instead? Unless I'm missing something, just when trying to follow the code.

Reply:
Yeah, I changed it to use a custom enum.